tests/lib/nrf_modem_lib/trace_backends/uart/src/main.c

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdbool.h>
+#include <string.h>
+
+#include <zephyr/kernel.h>
+
+#include <unity.h>
+
+#include "trace_backend.h"
+
+#include "mock_nrfx_uarte.h"
+#include "mock_pinctrl.h"
+
+extern int unity_main(void);
+
+static const nrfx_uarte_t *p_uarte_inst_in_use;
+/* Variable to store the event_handler registered by the modem_trace module.*/
+static nrfx_uarte_event_handler_t uarte_callback;
+
+/* Suite teardown shall finalize with mandatory call to generic_suiteTearDown. */
+extern int generic_suiteTearDown(int num_failures);
```

Comment:
nit: keep the `extern` prototypes together, move close to `extern int unity_main(void)`